### PR TITLE
MetricsBasedResizerSpec appears to be at risk of race condition failure

### PR DESCRIPTION
**Pull request by markalanrichards**
Got a test error running the complete suite of tests and other tests seem to have this similar Thread sleep.
Please check and accept if appropriate.
Thanks

### CODE REVIEW

File: akka-actor-tests/src/test/scala/akka/routing/MetricsBasedResizerSpec.scala [hunk 1]
```diff
@@ -212,6 +212,7 @@ class MetricsBasedResizerSpec extends AkkaSpec(ResizerSpec.config) with DefaultT
       msgs.head.open()
 
       router.sendToAll()
+      Thread.sleep(1) // wait for routees to update their mail boxes
```
Comment (patriknw) [inline akka-actor-tests/src/test/scala/akka/routing/MetricsBasedResizerSpec.scala]: I would prefer to put this in an `awaitAssert` instead

```scala
awaitAssert {
  resizer.reportMessageCount(router.routees, router.msgs.size)
  resizer.performanceLog.get(2) should not be empty
}
```

Comment (viktorklang) [inline akka-actor-tests/src/test/scala/akka/routing/MetricsBasedResizerSpec.scala]: :+1: @patriknw

**Comment by markalanrichards** [comment]
Sure, I'm not too worried whether it is Thread.sleep or awaitAssert, I was following what looked like a habit in that area of the tests: does this mean the other tests will be swapped to awaitAssert too?

**Comment by patriknw** [comment]
if possible, that would be good

**Comment by markalanrichards** [comment]
Great

File: akka-actor-tests/src/test/scala/akka/routing/MetricsBasedResizerSpec.scala (hunk 1 above)
Comment (drewhk) [inline akka-actor-tests/src/test/scala/akka/routing/MetricsBasedResizerSpec.scala]: agree

**Comment by drewhk** [comment]
@markalanrichards  do you have time to finish this, or would you prefer if we take over?

**Comment by rkuhn** [comment]
I’m taking this over now.

**Comment by rkuhn** [comment]
Thanks for noticing, I ended up making it deterministic without sleeps :-)

**Pull request closed without merging**